File: FMAplication/Domain/Transaction/Transaction.cs

using FMAplication.Core;
using FMAplication.Enumerations;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using FMAplication.Domain.Sales;
using FMAplication.Extensions;

namespace FMAplication.Domain.Transaction
{
    public class Transaction : AuditableEntity<int>
    {
        public string TransactionNumber { get; set; }
        public string Remarks { get; set; }
        public bool IsConfirmed { get; set; }
        public TransactionStatus TransactionStatus { get; set; }
        public TransactionType TransactionType { get; set; }
        public DateTime TransactionDate { get; set; }
        [NotMapped] 
        public string TransactionDateStr => TransactionDate.ToIsoString();
        public string ChalanNumber { get; set; }

        public WareHouse.WareHouse WareHouse { get; set; }  
        public int? WarehouseId { get; set; }
        
        
        public int SalesPointId { get; set; }
        public int ReferenceTransactionId { get; set; }

        public List<WDistributionTransaction> WDistributionTransactions { get; set; }
        public List<WDistributionRecieveTransaction> WDistributionRecieveTransactions { get; set; }
        public List<SalesPointAdjustmentItem> SalesPointAdjustmentItems { get; set; }
        public List<StockAdjustmentItems> StockAdjustmentItems { get; set; }

    }
}

File: FMAplication/ClientApp/src/app/Shared/Entity/Inventory/Transaction.ts
import { SalesPoint, UserInfo, WDistributionTransaction } from '..';
import { TransactionStatus } from '../../Enums/TransactionStatus';
import { TransactionType } from '../../Enums/TransactionType';
import { TransactionNotification } from '../Notification/TransactionNotification';
import { SalesPointStockAdjustmentItem } from '../salesPointStockAdjustment';
import { TransactionWorkflow } from '../TransactionWorkflow/TransactionWorkflow';
import { StockAddTransaction } from './StockAddTransaction';
import { WareHouse } from './WareHouse';
import { WDistributionRecieveTransaction } from './wDistributionRecieveTransaction';

export class Transaction {
    public id: number;
    createdBy:number;
    createdByUser:UserInfo;
    referenceTransactionId:number;
    referenceTransaction:Transaction;
    public transactionSerial: number;
    public remarks: string; 
    public isConfirmed: boolean=false; 
    public transactionStatus: TransactionStatus; 
    public transactionType: TransactionType;
    public transactionDate: string=new Date().toISOString();
    transactionDateStr:string = new Date().toISOString();;
    public transactionNumber: string=""; 
    public chalanNumber: string; 
    public wareHouseModel: WareHouse;
    public warehouseId: number; 
    public salesPoint: SalesPoint;
    public salesPointId: number;
    transactionWorkflow:TransactionWorkflow;
    transactionNotification:TransactionNotification
    public stockAddTransactions: StockAddTransaction[]=[];
    public wDistributionTransactions: WDistributionTransaction[]=[];
    wDistributionRecieveTransactions:WDistributionRecieveTransaction[]=[];
    salesPointAdjustmentItems:SalesPointStockAdjustmentItem[]=[];

}